core/drivers/crypto/caam/acipher/caam_ecc.c

```diff
@@ -0,0 +1,724 @@
+	} else {
+		ECC_TRACE("CAAM Status 0x%08" PRIx32, jobctx.status);
+		ret = job_status_to_tee_result(jobctx.status);
+	}
+
+out:
+	caam_free_desc(&desc);
+	do_keypair_free(&ecckey);
+	caam_dmaobj_free(&msg);
+	caam_dmaobj_free(&sign_d);
+	caam_dmaobj_free(&sign_c);
+
+	return ret;
+}
+
+/*
+ * Verification of the Signature of ECC message
+ * Note the message is already hashed
+ *
+ * @sdata   [in/out] ECC Signature to verify
+ */
+static TEE_Result do_verify(struct drvcrypt_sign_data *sdata)
+{
+	TEE_Result ret = TEE_ERROR_GENERIC;
+	enum caam_status retstatus = CAAM_FAILURE;
+	enum caam_ecc_curve curve = CAAM_ECC_UNKNOWN;
+	struct ecc_public_key *inkey = sdata->key;
+	struct caam_ecc_keypair ecckey = { };
+	struct caambuf tmp = { };
+	struct caam_jobctx jobctx = { };
+	uint32_t *desc = NULL;
+	uint32_t desclen = 0;
+	struct caamdmaobj msg = { };
+	struct caamdmaobj sign_c = { };
+	struct caamdmaobj sign_d = { };
+	uint32_t pdb_sgt_flags = 0;
+
+#ifdef CFG_CAAM_64BIT
+#define MAX_DESC_VERIFY 15
+#else
+#define MAX_DESC_VERIFY 10
+#endif
+
+	ECC_TRACE("ECC Verify");
+
+	/* Verify first if the curve is supported */
+	curve = get_caam_curve(inkey->curve);
+	if (curve == CAAM_ECC_UNKNOWN)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	/* Allocate the job descriptor */
+	desc = caam_calloc_desc(MAX_DESC_VERIFY);
+	if (!desc) {
+		ret = TEE_ERROR_OUT_OF_MEMORY;
+		goto out;
+	}
+
+	/* Convert the Public key to local key */
+	retstatus = do_keypub_conv(&ecckey, inkey, sdata->size_sec);
+	if (retstatus != CAAM_NO_ERROR) {
+		ret = caam_status_to_tee_result(retstatus);
+		goto out;
+	}
+
+	/* Prepare the input message CAAM Descriptor entry */
+	ret = caam_dmaobj_input_sgtbuf(&msg, sdata->message.data,
+				       sdata->message.length);
+	if (ret)
+		goto out;
+
+	if (msg.sgtbuf.sgt_type)
+		pdb_sgt_flags |= PDB_SGT_PKVERIF_MSG;
+
+	caam_dmaobj_cache_push(&msg);
+
+	/*
+	 * Prepare the 1st Part of the signature
+	 * Handle the full signature in case signature buffer needs to
+	 * be reallocated.
+	 */
+	ret = caam_dmaobj_input_sgtbuf(&sign_c, sdata->signature.data,
+				       sdata->signature.length);
+	if (ret)
+		goto out;
+
+	if (sign_c.sgtbuf.sgt_type)
+		pdb_sgt_flags |= PDB_SGT_PKVERIF_SIGN_C;
+
+	/* Prepare the 2nd Part of the signature, derived from sign_c */
+	ret = caam_dmaobj_derive_sgtbuf(&sign_d, &sign_c, sdata->size_sec,
+					sdata->size_sec);
+	if (ret)
+		goto out;
+
+	if (sign_d.sgtbuf.sgt_type)
+		pdb_sgt_flags |= PDB_SGT_PKVERIF_SIGN_D;
+
+	caam_dmaobj_cache_push(&sign_c);
+
+	/* Allocate a Temporary buffer used by the CAAM */
+	retstatus = caam_alloc_align_buf(&tmp, 2 * sdata->size_sec);
+	if (retstatus != CAAM_NO_ERROR) {
+		ret = caam_status_to_tee_result(retstatus);
+		goto out;
+	}
+
+	/* Build the descriptor using Predifined ECC curve */
+	caam_desc_init(desc);
+	caam_desc_add_word(desc, DESC_HEADER(0));
+	caam_desc_add_word(desc, PDB_PKVERIFY_PD1 | PDB_ECC_ECDSEL(curve)
+				 | pdb_sgt_flags);
```

Comment:
Please begin the new line with the value, that is, leave the operation at the end of the line above.

Reply:
Ok